StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory.swift

```diff
@@ -643,20 +643,33 @@ extension PaymentSheetFormFactory {
         return StaticElement(view: label)
     }
 
-    func makeInstantDebits() -> PaymentMethodElement {
+    func makeInstantDebits(countries: [String]? = nil) -> PaymentMethodElement {
+        let titleElement: StaticElement? = if case .paymentSheet = configuration {
+            makeSectionTitleLabelWith(text: Self.PayByBankDescriptionText)
+        } else {
+            nil
+        }
+
+        let billingConfiguration = configuration.billingDetailsCollectionConfiguration
+        let nameElement = billingConfiguration.name == .always ? makeName() : nil
+        let phoneElement = billingConfiguration.phone == .always ? makePhone() : nil
+        let addressElement = billingConfiguration.address == .full
+        ? makeBillingAddressSection(collectionMode: .all(), countries: countries)
+            : nil
+
+        // An email is required, so only hide the email field iff:
+        // The configuration specifies never collecting email, and a default (non-empty) email is provided.
+        let shouldHideEmailField = billingConfiguration.email == .never &&
```

Comment:
We changed the logic [on Android](https://github.com/stripe/stripe-android/blob/f361214ac98d8b79d68c7845dc123ce10b4c9679/paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/AddPaymentMethodRequirement.kt#L87) for this: We don’t show the Bank tab if the merchant has requested to never collect email, didn’t provide a default value.

This means that the check here can be changed, and we’ll need to add the check in `filteredPaymentMethodTypes()`.

Reply:
Done in latest commit: https://github.com/stripe/stripe-ios/pull/4162/commits/ade9156c3e01607e86e0c0ede53ec9c5f23c2826

https://github.com/user-attachments/assets/6c282ec5-f23c-438d-9947-7b1d3f07cfda